onnx/test/shape_inference_test.py

```diff
@@ -174,6 +192,65 @@ def _compare_value_infos(
             )
 
 
+def single_op_all_version_tester(op_name: str):
+    """This is decorator to run one test case for all versions of an Op.
+
+    However, there is a creation issue for Reshape Operator under version 5, which
+    probably caused by the mismatch creation arguments in _make_graph.
+    Therefore, all test cases now only run when version > 5.
+    Issue reference: https://github.com/onnx/onnx/issues/5289.
+
+    Args:
+        op_name: The name of the Op, which should be in `defs.get_all_schemas`.
+
+    Returns:
+        The decorated function for unittest case.
+        For example, we can use:
+        ```
+        @single_op_all_version_tester("Transpose")
+        def test_transpose(self, version):
+            ...
+        ```
+
+    Raises:
+        KeyError: If the op_name is not in `defs.get_all_schemas`.
+    """
+    cls = TestShapeInferenceHelper
+    if len(cls.all_op_versions) == 0:
+        # Init the version map for all Ops.
+        for schema in defs.get_all_schemas():
+            cls.all_op_versions[schema.name] = cls.get_available_versions(schema)
+
+    def decorator(test_function):
+        if op_name not in cls.all_op_versions:
+            raise KeyError(f"Invalid Op name: {op_name}")
+        versions = cls.all_op_versions[op_name]
+        assert versions is not None
+
+        def wrapper(*args, **kwargs):
+            assert len(versions) > 0
+            for version in versions:
+                # Many versions found for this op.
+                if int(version) <= 5:
```

Comment:
This could lead to a disabled test if all possible version are <= 5.

Reply:
Yes, I added an issue to track this. https://github.com/onnx/onnx/issues/5289
Currently, shape_inference_test.py only test recent op version by default. This PR helps to enable tests for more op versions > 5.